io/src/vlp_grabber.cpp

```diff
@@ -205,3 +223,14 @@ pcl::VLPGrabber::getName () const
   return (std::string ("Velodyne LiDAR (VLP) Grabber"));
 }
 
+/////////////////////////////////////////////////////////////////////////////
+void
+pcl::VLPGrabber::setLaserColorRGB (const pcl::RGB& color,
+                                   unsigned int laserNumber)
+{
+  if (laserNumber >= (unsigned int) VLP_MAX_NUM_LASERS)
+      return;
+
+  laser_rgb_mapping_[laserNumber] = color;
+}
```

Comment:
It's kinda limiting to only be able to set one color at the time on a 16 element array. Why not add something which allows copying an entire block?

Reply:
Does that mean adding a function such as following?
```cpp
// set one color to all lasers
void
pcl::VLPGrabber::setLaserColorRGB (const pcl::RGB& color)
{
  for (uint8_t i = 0; i < VLP_MAX_NUM_LASERS; i++)
  {
    setLaserColorRGB (color, i); 
  } 
}
```
or
```cpp
// set colors array to lasers
void
pcl::VLPGrabber::setLaserColorRGB (const pcl::RGB (&colors)[VLP_MAX_NUM_LASERS])
{
  laser_rgb_mapping_ = colors;
}
```

Reply:
I think it should be added to the base class as well as this class.
These change would be better to separate it into other pull requests.

Reply:
> Does that mean adding a function such as following?

I meant the second one but if you want to be generic between HDL and VPL grabbers we should provide a signature which specifies iterators for the read beginning and end positions.

> I think it should be added to the base class as well as this class.
These change would be better to separate it into other pull requests.

No need for the separate PR. Just add another commit providing the setters in HDL.

Reply:
> I meant the second one but if you want to be generic between HDL and VPL grabbers we should provide a signature which specifies iterators for the read beginning and end positions.

Can you show implementation examples?

> No need for the separate PR. Just add another commit providing the setters in HDL.

I get it.

Reply:
Something like the following would be ok for both base and derived classes but won't allow you to make a boundary check
```cpp
// provide iteraror range
template<typename IterT> void
pcl::HDLGrabber::setLaserColorRGB (const IterT& begin, const IterT& end)
{
  std::copy (begin, end, laser_rgb_mapping_);
}
```
This needs to be implemented in the header because of the template and HDL will require it's own version due to the different name of the constant. 

Another idea to allow boundary checking would be to rename these max num lasers constants to a single max_num_lasers, which is set on object construction, but this already involves a lot more changes...